static/app/utils/redirect.tsx

```diff
@@ -1,17 +1,23 @@
 import {useEffect} from 'react';
-import {InjectedRouter} from 'react-router';
+import {browserHistory, InjectedRouter} from 'react-router';
 import {LocationDescriptor} from 'history';
 
 type Props = {
-  router: InjectedRouter;
   to: LocationDescriptor;
+  router?: InjectedRouter;
 };
 
 // This is react-router v4 <Redirect to="path/" /> component to allow things
 // to be declarative.
 function Redirect({to, router}: Props) {
   // Redirect on mount.
-  useEffect(() => router.replace(to), []);
+  useEffect(() => {
+    if (router) {
+      router.replace(to);
+    } else {
+      browserHistory.replace(to);
```

Comment:
@Neo-Zhixing instead of using `browserHistory` directly here, could `useNavigate()` be useful as an alternative?

@NisanthanNanthakumar has introduced some react router hooks in https://github.com/getsentry/sentry/pull/32813